turbinia/workers/file_system_timeline.py

```diff
@@ -67,25 +68,42 @@ def run(self, evidence, result):
     # Iterate over all file entries and generate the output in bodyfile
     # format.
     try:
-      with open(bodyfile_output, 'w') as file_object:
-        for file_entry, path_segments in entry_lister.ListFileEntries(
-            base_path_specs):
-          bodyfile_entries = entry_lister.GetBodyfileEntries(
-              file_entry, path_segments)
-          for bodyfile_entry in bodyfile_entries:
-            file_object.write(bodyfile_entry)
-            file_object.write('\n')
-            number_of_entries += 1
-      output_evidence.number_of_entries = number_of_entries
-      result.add_evidence(output_evidence, evidence.config)
-      status = 'Generated file system timeline containing [{0:d}] entries'.format(
-          number_of_entries)
-      result.close(self, success=True, status=status)
-    except dfvfs_errors.ScannerError as exception:
-      result.log('Error generating bodyfile {0!s}'.format(exception))
-      status = 'Unable to generate bodyfile using provided evidence data.'
+      file_entries = None
+      with open(bodyfile_output, 'w', encoding='utf-8') as file_object:
+        try:
+          file_entries = enumerate(
+              entry_lister.ListFileEntries(base_path_specs))
+          while file_entries:
+            try:
+              _, (file_entry, path_segments) = next(file_entries)
+              bodyfile_entries = entry_lister.GetBodyfileEntries(
+                  file_entry, path_segments)
+              for bodyfile_entry in bodyfile_entries:
+                file_object.write(bodyfile_entry)
+                file_object.write('\n')
+                number_of_entries += 1
+            except StopIteration:
+              break
+        except (dfvfs_errors.AccessError, dfvfs_errors.BackEndError,
+                dfvfs_errors.MountPointError,
+                dfvfs_errors.PathSpecError) as exception:
+          status = 'Unable to process file entry: {0!s}'.format(exception)
+          result.log(status)
+
+      if number_of_entries > 0:
+        output_evidence.number_of_entries = number_of_entries
+        result.add_evidence(output_evidence, evidence.config)
+        status = 'Generated file system timeline containing [{0:d}] entries'.format(
+            number_of_entries)
+        result.close(self, success=True, status=status)
+      else:
+        status = 'Unable to process any file entries: {0!s}'.format(exception)
```

Comment:
I don't think we have `exception` defined at this scope, so we probably need to update this status string.

Reply:
Fixed.